cycles-ledger/tests/tests.rs

```diff
@@ -2935,6 +2965,564 @@ fn test_icrc1_transfer_duplicate(env: &TestEnv) {
+        for should_set_expires_at in [DontSetExpiresAt, SetExpiresAt] {
+            for should_set_fee in [DontSetFee, SetFee] {
+                for should_set_memo in [DontSetMemo, SetMemo] {
+                    test_icrc2_approve_ok_with_params(
+                        env,
+                        DontSetCreatedAtTime,
+                        should_set_fee,
+                        should_set_memo,
+                        should_set_expected_allowance,
+                        should_set_expires_at,
+                    )
+                }
+            }
+        }
+    }
+}
+
+fn test_icrc2_approve_ok_with_created_at_time(env: &TestEnv) {
+    use ShouldSetCreatedAtTime::*;
+    use ShouldSetExpectedAllowance::*;
+    use ShouldSetExpiresAt::*;
+    use ShouldSetFee::*;
+    use ShouldSetMemo::*;
+
+    for should_set_expected_allowance in [DontSetExpectedAllowance, SetExpectedAllowance] {
+        for should_set_expires_at in [DontSetExpiresAt, SetExpiresAt] {
+            for should_set_fee in [DontSetFee, SetFee] {
+                for should_set_memo in [DontSetMemo, SetMemo] {
+                    test_icrc2_approve_ok_with_params(
+                        env,
+                        SetCreatedAtTime,
+                        should_set_fee,
+                        should_set_memo,
+                        should_set_expected_allowance,
+                        should_set_expires_at,
+                    )
+                }
+            }
+        }
+    }
+}
+
+fn test_icrc2_approve_from_denied_owner(env: &TestEnv) {
+    let account_spender = account(3, None);
+    let account_spender_balance = env.icrc1_balance_of(account_spender);
+    let total_supply = env.icrc1_total_supply();
+    let blocks = env.get_all_blocks();
+    for owner in [Principal::anonymous(), Principal::management_canister()] {
+        for subaccount in [None, Some([0; 32])] {
+            let args = ApproveArgs {
+                from_subaccount: subaccount,
+                spender: account_spender,
+                amount: Nat::from(0u8),
+                fee: None,
+                created_at_time: None,
+                memo: None,
+                expected_allowance: None,
+                expires_at: None,
+            };
+            let expected_error = ApproveError::GenericError {
+                error_code: Nat::from(DENIED_OWNER),
+                message: format!(
+                    "Owner of the account {} cannot be part of approvals",
+                    Account { owner, subaccount },
+                ),
+            };
+            assert_eq!(Err(expected_error), env.icrc2_approve(owner, args),);
+        }
+    }
+    assert_eq!(
+        account_spender_balance,
+        env.icrc1_balance_of(account_spender)
+    );
+    assert_eq!(total_supply, env.icrc1_total_supply());
+    assert_vec_display_eq(blocks, env.get_all_blocks());
+}
+
+fn test_icrc2_approve_spender_denied_owner(env: &TestEnv) {
```

Comment:
```suggestion
fn test_icrc2_approve_from_denied_spender(env: &TestEnv) {
```

Reply:
Done with e4dc151, I've also changed the other tests that were using the same schema (e.g. `test_icrc1_transfer_from_denied_owner`).